packages/vite/src/node/plugins/css.ts

```diff
@@ -522,13 +522,13 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
         if (isDirectCSSRequest(id)) {
           return null
         }
-        // server only
-        if (this.environment.config.consumer !== 'client') {
-          return modulesCode || `export default ${JSON.stringify(css)}`
-        }
         if (inlined) {
           return `export default ${JSON.stringify(css)}`
         }
+        // server only
+        if (this.environment.config.consumer !== 'client') {
```

Comment:
I think using `=== 'server'` here is clear without the comment
```suggestion
        if (this.environment.config.consumer === 'server') {
```